Further improvements on the application form

viktorcsimma opened the pull request:
- There is now no default resident/extern status. However, both the finalization pages and the radio button itself requires a selection.
- Reworked the disclaimer for the Informatics workshop; the most notable change is that on the finalization page, it appears only for users who have indeed applied to the workshop.
- Language exams can be deleted.
- Other small changes.

viktorcsimma committed:
Further improvements on the application form

deepsource-autofix[bot] committed:
style: format code with PHP CS Fixer

This commit fixes the style issues introduced in 7c38902 according to the output
from PHP CS Fixer.

Details: https://github.com/EotvosCollegium/mars/pull/606

BertalanD committed:
Merge branch 'development' into application-improvements

BertalanD commented:
hmm, the database seeder now seems to generate a null `applied_for_resident_status` for all aplicants.

BertalanD commented on resources/views/user/alfonso-language-exams.blade.php:
```diff
@@ -1,22 +1,44 @@
+                        <td>{{ $exam->level }}</td>
+                        <td>{{ $exam->type }}</td>
+                        <td>
+                            @if($exam->wasBeforeEnrollment and !isset($application))
+                            {{$exam->date->format('Y-m')}} (collegista státusz előtt szerezve)
+                            @else
+                            {{$exam->date->format('Y-m')}}
+                            @endif
```
nit: now that this is a full-width table, we could maybe also include the day (the day being chopped off kind of seems like a bug)

viktorcsimma replied:
Right; done:)

viktorcsimma commented:
Oh; I've missed that... I'll fix it. Thanks!

BertalanD commented on app/Http/Controllers/Secretariat/UserController.php:
```diff
@@ -296,6 +297,28 @@ public function uploadLanguageExam(Request $request, User $user)
+    public function deleteLanguageExam(Request $request, User $user, LanguageExam $exam)
+    {
+        $this->authorize('view', $user);
+        if ($exam->educationalInformation->user->id != $user->id) {
+            abort(400, 'The language exam does not belong to the given user.');
+        }
```
Not a PHP expert, feel free to disregard. But comparing database keys seems a bit fragile. Maybe something like

```suggestion
        if ($exam->educationalInformation->user->isNot($user)) {
            abort(400, 'The language exam does not belong to the given user.');
        }
```

viktorcsimma replied:
Nice catch:) I've not heard about this before.

viktorcsimma committed:
Fixing seeder generating null statuses for submitted applications

viktorcsimma commented:
Done;)

deepsource-autofix[bot] committed:
style: format code with PHP CS Fixer

This commit fixes the style issues introduced in 8d11cf5 according to the output
from PHP CS Fixer.

Details: https://github.com/EotvosCollegium/mars/pull/606

BertalanD commented on resources/views/auth/application/questions.blade.php:
```diff
@@ -48,15 +48,16 @@
                         <p>
-                            @php $checked = old('status') ?  old('status') == 'extern' : !$user->application->applied_for_resident_status @endphp
+                            @php $checked = old('status') ?  old('status') == 'extern'
+                                    : (!is_null($user->application->applied_for_resident_status) && !$user->application->applied_for_resident_status) @endphp
```
I _understand_ what this code is doing, but I refuse to believe that handling a three-state flag is this convoluted in PHP.

viktorcsimma replied:
I've looked it up now; it seems `===` does the job.

BertalanD reviewed (commented):
looks good overall, a couple things seemed odd to my (untrained) eye; feel free to ignore if they're not relevant

viktorcsimma commented on database/seeders/UsersTableSeeder.php:
```diff
@@ -132,7 +132,20 @@ private function createCollegist($user)
-        $user->application()->create(['submitted' => rand(0, 1)]);
+        $submitted = rand(0, 1);
+        // submitted applications must not have a null status
+        if ($submitted) {
+            $appliedForResidentStatus = rand(0, 1);
+        } else {
+            $appliedForResidentStatus = rand(-1, 1);
+            if (-1 == $appliedForResidentStatus) {
+                $appliedForResidentStatus = null;
+            }
+        }
```
I'll add some more comments. Thanks!

viktorcsimma committed:
Applying suggestions

viktorcsimma commented:
Nice suggestions; thanks:) I've applied all of them.
